### DISCUSSION TITLE
【UE】C++的模板绑定不支持UE的类型结构么？

### DISCUSSION THREAD

**Discussion by watsonsong** [Tencent/puerts#1476] [Q&A]
我在使用C++的[模板绑定](https://puerts.github.io/en/docs/puerts/unreal/template_binding/)的时候，如果在绑定函数的参数，返回值中出现了F结构体，A和U的类型，都会报编译错误：Error C2039 : "value": 不是 "puerts::ScriptTypeName<T,void>" 的成员。
例如std::function来作为一个回调注册给TS，只要回调函数的参数中有一个F结构体的话，就会编译不过。
这个可以通过某种扩展支持么，还是设计上不允许这样？

**Comment by chexiongsheng** [comment]
看文档，看例子。你应该是没声明对。
puerts本来很多UE的API就是用静态模板绑定

**Comment by watsonsong** [comment]
确实是使用问题，之前是因为没有给所有使用到的类型，包括返回值和参数都使用UsingCppType，UsingUStruct或者UsingUClass去定义类型。
